src/api/bulk.ts

```diff
@@ -1234,34 +1250,39 @@ export class QueryJobV2<S extends Schema> extends EventEmitter {
   }
 
   private getResultsUrl() {
-    const url = `${this.#connection.instanceUrl}/services/data/v${
-      this.#connection.version
-    }/jobs/query/${getJobIdOrError(this.jobInfo)}/results`;
-
-    return this.locator ? `${url}?locator=${this.locator}` : url;
-  }
+    const url = new URL(
+      `${this.#connection.instanceUrl}/services/data/v${
+        this.#connection.version
+      }/jobs/query/${getJobIdOrError(this.jobInfo)}/results`,
+    );
 
-  async getResults(): Promise<Record[]> {
-    if (this.finished && this.#queryResults) {
-      return this.#queryResults;
+    if (this.locator) {
+      url.searchParams.set('locator', this.locator);
+    }
+    if (this.#maxRecords) {
+      url.searchParams.set('maxRecords', this.#maxRecords.toString());
     }
 
-    this.#queryResults = [];
+    return url.toString();
+  }
 
-    while (this.locator !== 'null') {
-      const nextResults = await this.request<Record[]>({
+  private async *getResults(): AsyncGenerator<Record[], void, unknown> {
```

Comment:
Keep in mind, I am no longer a maintainer on this project, so I would leave it up to @stomita or @mshanemc

Reply:
I removed the `getRecords()` because [bulk v1 ](https://jsforce.github.io/document/#bulk-query)does not have it, and because I wanted to make v2's API equal to v1. Another good reason for creating this method, is because it could cause  `Fatal error: Javascript heap out of memory` if used to query huge data sets without using the LIMIT keyword.

Reply:
@AllanOricil Even though jsforce v2 is still in beta there's lot of projects using it so I would prefer not break the current bulk v2 implementation. What do you think about deprecating this and add an option to the query job, then make that the default at v2 GA?

maybe a third arg to `BulkV2.query` that makes it return the job instead of the results, default to false.